jkamz/ireporter
# #164325492 A user can receive verification link via email on successful signup

### What does this PR do?
Adds account activation for users via email upon successful registration

### Description of the task to be completed?
- Add email capability for sending verification links to registered users
- Add more fields to collect user's details
- Add tests for signup and activation endpoints

### How should you manually test this?
*Step 1*
 
Navigate to your working directory, create and activate the virtual environment 

```$  virtualenv -p python .venv```


```$ source .venv/bin/activate```

Clone the repository 

```$ git clone https://github.com/ann-mukundi/ireporter.git```

Switch to testing branch

```$ git checkout ft-signup-verify-email-164325492  ```

Switch to the ireporter directory

```$ cd ireporter/```

Install project dependencies

```$ pip install -r api/requirements.txt```

Run the database migrations

```$ python api/manage.py migrate```

*Step 2*

### Running the application

```$ python api/manage.py runserver```

### Testing

```$ python api/manage.py test```


### Prerequisites

Python, Git,  Virtualenv


### What are the relevant PT stories?
[#164325492](https://www.pivotaltracker.com/story/show/164325492)


### Screenshots
**Successful Signup**

![screenshot from 2019-03-07 07-48-17](https://user-images.githubusercontent.com/33033576/53938528-98ec4e80-40c1-11e9-9be0-d504f075881b.png)

**Successful Activation of account**

![screenshot from 2019-03-07 11-44-02](https://user-images.githubusercontent.com/33033576/53944316-2b93ea00-40d0-11e9-85a1-a741ea12daf6.png)


**Validations**
![screenshot from 2019-03-07 07-48-29](https://user-images.githubusercontent.com/33033576/53938548-a73a6a80-40c1-11e9-9a5c-d26a96641e39.png)
![screenshot from 2019-03-07 07-49-23](https://user-images.githubusercontent.com/33033576/53938560-b0c3d280-40c1-11e9-96ae-e1cd80c68f03.png)

**Invalid payload data**
![screenshot from 2019-03-07 07-46-55](https://user-images.githubusercontent.com/33033576/53938662-f08aba00-40c1-11e9-9c84-88c685cdf617.png)

![screenshot from 2019-03-07 07-46-36](https://user-images.githubusercontent.com/33033576/53938648-e5d02500-40c1-11e9-9009-bafc27ddfd55.png)

**Already Activated Account**
![screenshot from 2019-03-07 07-47-50](https://user-images.githubusercontent.com/33033576/53938686-0ac49800-40c2-11e9-833c-ec73078f77ea.png)
